src/scripts/main.js

```diff
@@ -1,3 +1,147 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table tbody');
+const headers = document.querySelectorAll('th');
+const formContainer = document.createElement('div');
+
+formContainer.innerHTML = `
+  <form class="new-employee-form">
+    <label>Name: <input name="name" type="text" data-qa="name" required></label>
+    <label>Position: <input name="position" type="text" data-qa="position" required></label>
+    <label>Office:
+      <select name="office" data-qa="office" required>
+        <option value="Tokyo">Tokyo</option>
+        <option value="Singapore">Singapore</option>
+        <option value="London">London</option>
+        <option value="New York">New York</option>
+        <option value="Edinburgh">Edinburgh</option>
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <label>Age: <input name="age" type="number" data-qa="age" required></label>
+    <label>Salary: <input name="salary" type="number" data-qa="salary" required></label>
+    <button type="submit">Save to table</button>
+  </form>
+`;
+
+document.body.prepend(formContainer);
+
+const showNotification = (type, title, description) => {
+  const notification = document.createElement('div');
+
+  notification.className = `notification ${type}`;
+  notification.setAttribute('data-qa', 'notification');
+  notification.innerHTML = `<h2 class="title">${title}</h2><p>${description}</p>`;
+  document.body.appendChild(notification);
+
+  setTimeout(() => notification.remove(), 2000);
+};
+
+let currentSort = {
+  column: null,
+  direction: 'asc',
+};
+
+headers.forEach((header, index) => {
+  header.addEventListener('click', () => {
+    const rows = Array.from(table.rows);
+    const newDirection =
+      currentSort.column === index && currentSort.direction === 'asc'
+        ? 'desc'
+        : 'asc';
+
+    currentSort = { column: index, direction: newDirection };
+
+    rows.sort((a, b) => {
+      const aText = a.cells[index].innerText;
+      const bText = b.cells[index].innerText;
+
+      if (newDirection === 'asc') {
+        return aText.localeCompare(bText, undefined, { numeric: true });
+      } else {
+        return bText.localeCompare(aText, undefined, { numeric: true });
```

Comment:
The sorting logic uses `localeCompare`, which may not correctly sort numeric values if they are not formatted as strings. Consider converting the cell values to numbers for numeric columns before comparing.